assets/quick-add.js

```diff
@@ -0,0 +1,79 @@
+if (!customElements.get('quick-add-modal')) {
+  customElements.define('quick-add-modal', class QuickAddModal extends ModalDialog {
+    constructor() {
+      super();
+      this.modalContent = this.querySelector('[id^="QuickAddInfo-"]');
+    }
+
+    hide(preventFocus = false) {
+      const cartNotification = document.querySelector('cart-notification');
+      if (cartNotification) cartNotification.setActiveElement(this.openedBy);
+      this.modalContent.innerHTML = '';
+      
+      if (preventFocus) this.openedBy = null;
+      super.hide();
+    }
+
+    show(opener) {
+      opener.setAttribute('aria-disabled', true);
+      opener.classList.add('loading');
+      opener.querySelector('.loading-overlay__spinner').classList.remove('hidden');
+
+      fetch(opener.getAttribute('data-product-url'))
+        .then((response) => response.text())
+        .then((responseText) => {
+          const responseHTML = new DOMParser().parseFromString(responseText, 'text/html');
+          const productInfo = responseHTML.querySelector('section[id^="MainProduct-"]');
+
+          this.setInnerHTML(this.modalContent, this.removeDOMElements(productInfo).innerHTML);
+          
+          if (window.Shopify && Shopify.PaymentButton) {
+            Shopify.PaymentButton.init();
+          }
+
+          if (window.ProductModel) window.ProductModel.loadShopifyXR();
+          
+          this.removeGalleryListSemantic();
+          this.preventVariantURLSwitching();
+          super.show(opener);
+        })
+        .finally(() => {
+          opener.removeAttribute('aria-disabled');
+          opener.classList.remove('loading');
+          opener.querySelector('.loading-overlay__spinner').classList.add('hidden');
+        });
+    }
+
+    setInnerHTML(element, html) {
+      element.innerHTML = html;
+      element.querySelectorAll("script").forEach( oldScript => {
+        const newScript = document.createElement("script");
+        Array.from(oldScript.attributes).forEach(attribute => newScript.setAttribute(attribute.name, attribute.value));
+        newScript.appendChild(document.createTextNode(oldScript.innerHTML));
+        oldScript.parentNode.replaceChild(newScript, oldScript);
+      });
+    }
```

Comment:
This is an obscure snippet to solve the obscure problem of scripts not executing when copying innerHTML. Could we leave a small comment to explain this? I don't think it would be clear why it's needed.

There's also a couple small formatting things like the unneeded space before oldScript and the use of double quotes instead of single.

Reply:
Done! :D